mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,195 @@
+        pyplot.clf()
+        do_plot()
+        artifact_file_name = f"{artifact_name}_on_{dataset_name}.png"
+        artifact_file_local_path = temp_dir.path(artifact_file_name)
+        pyplot.savefig(artifact_file_local_path)
+        mlflow.log_artifact(artifact_file_local_path, artifact_file_name)
+        artifacts[artifact_file_name] = \
+            ImageEvaluationArtifact(
+                location=get_artifact_uri(run_id, artifact_file_name),
+                content=ImageEvaluationArtifact.load_content_from_file(artifact_file_local_path)
+            )
+
+    def _log_model_explainality(
+        self, artifacts, temp_dir, model, X, dataset_name, run_id
+    ):
+        explainer = shap.Explainer(model, X)
+        shap_values = explainer(X)
+
+        def plot_summary():
+            shap.plots.beeswarm(shap_values, show=False)
+
+        self._log_image_artifact(artifacts, temp_dir, plot_summary, run_id, 'shap_summary', dataset_name)
+
+        def plot_feature_importance():
+            shap.plots.bar(shap_values, show=False)
+
+        self._log_image_artifact(
+            artifacts, temp_dir, plot_feature_importance, run_id, 'shap_feature_importance', dataset_name
+        )
+
+    def _classifier_compute_metrics_and_compute_and_log_artifacts(
+        self, temp_dir, model, X, y, dataset_name, run_id, evaluator_config
+    ):
+        # TODO: how to get num_classes without extra config ?
+        num_classes = evaluator_config['num_classes']
+        y_pred = model.predict(X)
+
+        is_binomial = (num_classes <= 2)
+
+        metrics = EvaluationMetrics()
+        artifacts = {}
+        metrics['accuracy'] = sk_metrics.accuracy_score(y, y_pred)
+        metrics['example_count'] = len(X)
+
+        # TODO: sum/mean on what data ?
+        #  [P0] Sum: Computes the (weighted) sum of the given values.
+        #  [P0] Mean: Computes the (weighted) mean of the given values.
+
+        if is_binomial:
+            if hasattr(model, 'predict_proba'):
+                y_probs = model.predict_proba(X)
+                y_prob = y_probs[:, 1]
+            else:
+                y_probs = None
+                y_prob = None
+
+            confusion_matrix = sk_metrics.confusion_matrix(y, y_pred)
+            tn, fp, fn, tp = confusion_matrix.ravel()
+            metrics['true_negatives'] = tn
+            metrics['false_positives'] = fp
+            metrics['false_negatives'] = fn
+            metrics['true_positives'] = tp
+            metrics['recall'] = sk_metrics.recall_score(y, y_pred)
+            metrics['precision'] = sk_metrics.precision_score(y, y_pred)
+            metrics['f1_score'] = sk_metrics.f1_score(y, y_pred)
+
+            # TODO:
+            #  compute hinge loss, this requires calling decision_function of the model
+            #  e.g., see https://scikit-learn.org/stable/modules/generated/sklearn.svm.LinearSVC.html#sklearn.svm.LinearSVC.decision_function
```

Comment:
I think hinge loss is only for SVM models

Reply:
pasting here for reference purposes: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.hinge_loss.html

This also requires a list of label values from the training data (a set of unique label values for multiclass classification tasks) which follows on the above comment (https://github.com/mlflow/mlflow/pull/5092#discussion_r755686975) that in order to calculate this metric, we will need information from the training event to be persisted as part of the Evaluation artifact.

Reply:
Hinge loss also require the "decision_function" output. Current PyFuncModel does not support the decision_function interface